### PR TITLE
ES6: migrated "client"

### PR DESCRIPTION
**Pull request by herrmannplatz**

### CODE REVIEW

File: app/script/client/ClientRepository.js [hunk 1]
```diff
@@ -0,0 +1,726 @@
...
+
+    this.client_mapper = new z.client.ClientMapper();
+    this.clients = ko.pureComputed(() => this.self_user() ? this.self_user().devices() : []);
+    this.current_client = ko.observable();
+
+    amplify.subscribe(z.event.Backend.USER.CLIENT_ADD, this.map_self_client);
+    amplify.subscribe(z.event.Backend.USER.CLIENT_REMOVE, this.on_client_remove);
+    amplify.subscribe(z.event.WebApp.LIFECYCLE.ASK_TO_CLEAR_DATA, this.logout_client);
+    // todo: deprecated - remove when user base of wrappers version >= 2.12 is large enough
+    amplify.subscribe(z.event.WebApp.LOGOUT.ASK_TO_CLEAR_DATA, this.logout_client);
```
Comment (lipis) [inline app/script/client/ClientRepository.js]: maybe it's time?

File: app/script/client/ClientMapper.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+ *
+ */
+
+'use strict';
+
+window.z = window.z || {};
+window.z.client = z.client || {};
+
+z.client.ClientMapper = class ClientMapper {
+
```
Comment (gregor) [inline app/script/client/ClientMapper.js]: No empty line

File: app/script/client/ClientRepository.js [hunk 3]
```diff
@@ -0,0 +1,722 @@
...
+      return client_ets;
+    });
+  }
+
+  get_client_by_user_id_from_db(user_id) {
+    return this.client_service.load_all_clients_from_db()
+    .then(clients => {
+      return clients
+        .filter((client) => (z.client.Client.dismantle_user_client_id(client.meta.primary_key)).user_id === user_id)
+        .map((client) => client);
```
Comment (gregor) [inline app/script/client/ClientRepository.js]: Obsolete line of code

File: app/script/client/ClientMapper.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  constructor() {
+  }
```
Comment (lipis) [inline app/script/client/ClientMapper.js]: do we need an empty `constructor`?

File: app/script/client/ClientService.js [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+  /*
+  Deletes the temporary client of a user.
+  @param {string} client_id - ID of the temporary client to be deleted
+  @return {Promise} - Resolves once the deletion of the temporary client is complete
+  */
+  delete_temporary_client(client_id) {
+    return this.client.send_json({
+      url: this.client.create_url(`${z.client.ClientService.URL_CLIENTS}/${client_id}`),
+      type: 'DELETE',
+      data: {}});
```
Comment (lipis) [inline app/script/client/ClientService.js]:
```js
return this.client.send_json({
  url: this.client.create_url(`${z.client.ClientService.URL_CLIENTS}/${client_id}`),
  type: 'DELETE',
  data: {},
});
```

File: app/script/client/ClientRepository.js [hunk 6]
```diff
@@ -0,0 +1,722 @@
...
+        identifier = z.string.wire_windows;
+      } else {
+        identifier = z.string.wire_linux;
+      }
+      device_model = z.localization.Localizer.get_text(identifier);
+      if (!z.util.Environment.frontend.is_production()) {
+        device_model = `${device_model} (Internal)`;
+      }
+    } else {
+      if (client_type === z.client.ClientType.TEMPORARY) {
```
Comment (lipis) [inline app/script/client/ClientRepository.js]: could be combined `else if`

File: app/script/client/ClientRepository.js [hunk 7]
```diff
@@ -0,0 +1,722 @@
...
+      amplify.publish(z.event.WebApp.CLIENT.UPDATE, user_id, client_ets);
+      return client_ets;
+    });
+  }
+
+  get_client_by_user_id_from_db(user_id) {
+    return this.client_service.load_all_clients_from_db()
+    .then(clients => {
+      return clients.filter((client) => {
+        return (z.client.Client.dismantle_user_client_id(client.meta.primary_key)).user_id === user_id
```
Comment (lipis) [inline app/script/client/ClientRepository.js]: `;` (From Travis ❌ )

Comment (herrmannplatz) [reply app/script/client/ClientRepository.js]: 💩

**Pull request merged**